cmd/ui/src/views/Explore/ContextMenu/ContextMenuZoneManagementEnabled.tsx

```diff
@@ -117,69 +101,62 @@ const ContextMenu: FC<{
         <Dialog open={dialogOpen}>
             <Menu
-                open={contextMenu !== null}
-                anchorPosition={{ left: contextMenu?.mouseX || 0, top: contextMenu?.mouseY || 0 }}
+                open={menuPosition !== null}
+                anchorPosition={menuPosition!}
                 anchorReference='anchorPosition'
                 onClick={onClose}
                 keepMounted>
-                <MenuItem onClick={handleSetStartingNode}>Set as starting node</MenuItem>
-                <MenuItem onClick={handleSetEndingNode}>Set as ending node</MenuItem>
-                {checkPermission(Permission.GRAPH_DB_WRITE) && [
-                    <AssetGroupMenuItem
-                        key={tierZeroAssetGroup!.id}
-                        assetGroupId={tierZeroAssetGroup!.id}
-                        assetGroupName={tierZeroAssetGroup!.name}
-                        onAddNode={handleAddNode}
-                        removeNodePath={`/zone-management/details/tier/${tierZeroAssetGroup!.id}`}
-                        isCurrentMember={isNode(selectedItemQuery.data) && selectedItemQuery.data.isTierZero}
-                        onShowConfirmation={() => {
-                            setDialogOpen(true);
-                        }}
-                        onCancelConfirmation={() => {
-                            setDialogOpen(false);
-                        }}
-                        showConfirmationOnAdd
-                        confirmationOnAddMessage={`Are you sure you want to add this node to ${tierZeroAssetGroup!.name}? This action will initiate an analysis run to update group membership.`}
-                    />,
-                    <AssetGroupMenuItem
-                        key={ownedAssetGroup!.id}
-                        assetGroupId={ownedAssetGroup!.id}
-                        assetGroupName={ownedAssetGroup!.name}
-                        onAddNode={handleAddNode}
-                        removeNodePath={`/zone-management/details/label/${ownedAssetGroup!.id}`}
-                        isCurrentMember={isNode(selectedItemQuery.data) && selectedItemQuery.data.isOwnedObject}
-                    />,
-                ]}
-                <CopyMenuItem />
+                {edgeItem && <EdgeMenuItems id={edgeItem.id} pathfindingFilters={pathfindingFilters} />}
+
+                {nodeItem && <NodeMenuItems exploreParams={exploreParams} objectId={nodeItem.objectId} />}
+
+                {nodeItem &&
+                    isAssetGroupEnabled && [
+                        <AssetGroupMenuItem
+                            key={tierZeroAssetGroup!.id}
+                            assetGroupId={tierZeroAssetGroup!.id}
+                            assetGroupName={tierZeroAssetGroup!.name}
+                            onAddNode={handleAddNode}
+                            removeNodePath={`/zone-management/details/tier/${tierZeroAssetGroup!.id}`}
+                            isCurrentMember={nodeItem!.isTierZero}
```

Comment:
Are we able to remove the non-null assertion because we check if nodeItem is truethy on line 132?